server/src/main/java/org/opensearch/cluster/metadata/Metadata.java

```diff
@@ -1511,40 +1580,28 @@ public Metadata build() {
             }
 
-            SortedMap<String, IndexAbstraction> indicesLookup = Collections.unmodifiableSortedMap(buildIndicesLookup());
+            TimeValue startTime = TimeValue.timeValueNanos(System.nanoTime());
+            indicesLookup = Collections.unmodifiableSortedMap(buildIndicesLookup());
+            TimeValue endTime = TimeValue.timeValueNanos(System.nanoTime());
+            // Logging for testing only - will remove in future iterations
+            logger.info(
+                "rebuilt indicesLookupMap in {} ms, {} entries",
+                (endTime.nanos() - startTime.nanos()) / 1000000L,
+                indicesLookup != null ? indicesLookup.size() : 0
+            );
 
             validateDataStreams(indicesLookup, (DataStreamMetadata) customs.get(DataStreamMetadata.TYPE));
 
             // build all concrete indices arrays:
             // TODO: I think we can remove these arrays. it isn't worth the effort, for operations on all indices.
             // When doing an operation across all indices, most of the time is spent on actually going to all shards and
             // do the required operations, the bottleneck isn't resolving expressions into concrete indices.
-            String[] allIndicesArray = allIndices.toArray(Strings.EMPTY_ARRAY);
-            String[] visibleIndicesArray = visibleIndices.toArray(Strings.EMPTY_ARRAY);
-            String[] allOpenIndicesArray = allOpenIndices.toArray(Strings.EMPTY_ARRAY);
-            String[] visibleOpenIndicesArray = visibleOpenIndices.toArray(Strings.EMPTY_ARRAY);
-            String[] allClosedIndicesArray = allClosedIndices.toArray(Strings.EMPTY_ARRAY);
-            String[] visibleClosedIndicesArray = visibleClosedIndices.toArray(Strings.EMPTY_ARRAY);
-
-            return new Metadata(
-                clusterUUID,
-                clusterUUIDCommitted,
-                version,
-                coordinationMetadata,
-                transientSettings,
-                persistentSettings,
-                hashesOfConsistentSettings,
-                indices,
-                templates,
-                customs,
-                allIndicesArray,
-                visibleIndicesArray,
-                allOpenIndicesArray,
-                visibleOpenIndicesArray,
-                allClosedIndicesArray,
-                visibleClosedIndicesArray,
-                indicesLookup
-            );
+            this.allIndices = allIndices.toArray(new String[allIndices.size()]);
+            this.allOpenIndices = allOpenIndices.toArray(new String[allOpenIndices.size()]);
+            this.allClosedIndices = allClosedIndices.toArray(new String[allClosedIndices.size()]);
+            this.visibleIndices = visibleIndices.toArray(new String[visibleIndices.size()]);
+            this.visibleOpenIndices = visibleOpenIndices.toArray(new String[visibleOpenIndices.size()]);
+            this.visibleClosedIndices = visibleClosedIndices.toArray(new String[visibleClosedIndices.size()]);
```

Comment:
the local variables names and private variables are same and are confusing as their data types are different.